docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,185 @@
+- arch: https://wiki.archlinux.org/index.php/Go_package_guidelines
+
+
+The process for building a package is as follows:
+- make a source package
+- compile it to get binary packages.
+
+Only the binary package is used by the user for installation. It is not a recommended pratice to directly integrate binaries.
+
+Building the source package is the most difficult part. But once it is done, we can use tools such as [fpm](https://github.com/jordansissel/fpm/wiki) to convert it to other package formats.
+
+We could also use a tool such as [nfpm](https://github.com/goreleaser/nfpm) to build the deb and rpm formats. But we would then be limited to these two formats until nfpm provides others (or develop that part). We could also face difficulties with it as it is still in development and the maintainer warns that some features are not yet available.
+
+To generate the deb package, we could try the [dh-make-golang](https://github.com/Debian/dh-make-golang) tool.
+
+The package could then be distributed:
+- by direct download
+- from our own repository
+- in a user repository if it follows the guidelines
+
+For retrieving user inputs we should use debconf for Debian packaging. There are no debconf-like for rpm, so we would have to implement a post-install script for it.
+
+#### Agnostic Packaging
+There are [3 major agnostic packaging system](https://www.ostechnix.com/linux-package-managers-compared-appimage-vs-snap-vs-flatpak/) for linux: AppImage, FlatPak and Snap. As AppImage and FlatPak are more desktop application oriented, we choose to focus on Snap.
+
+##### Snap
+[Snaps](https://snapcraft.io/first-snap#go) are containerised software packages. They auto-update daily and work on a variety of Linux distributions. They also revert to the previous version if an update fails. This feature would make it necessary to find out how to implement the rollout versioning.
+
+From the [snap documentation](https://snapcraft.io/docs/go-applications), it seems pretty straightforward to package an application. Snaps are defined in a yaml file. Running an application as a service is done only by specifying "daemon: simple" in the application description. 
+This would make us save the work of building a storage node service.
+
+Snaps can then be published in the snapcraft [app store](https://snapcraft.io/). In the store, we would able to monitor the number of installed snaps. It is possible to [host our own store](https://ubuntu.com/blog/howto-host-your-own-snap-store) but that the snap daemon only handles one repository. Therefore, the use of Canonical's store seems mandatory. Snaps integrate well with [github](https://snapcraft.io/build).
+A snap inside the store can be published in multiple versions in different [channels](https://snapcraft.io/docs/channels).
+
+Snaps have been known for suffering a long start-up time, but it has been [improved](https://snapcraft.io/blog/snap-startup-time-improvements).
```

Comment:
Doesn't this only apply to GUI apps?

Reply:
I have been looking around a bit, but I don't see anything mentioning that it is only for GUIs, but it could be. Do you have reference on that?